server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRestoreTask.java

```diff
@@ -56,6 +60,18 @@ public String getUser() {
             return user.asString();
         }
 
+        public String getRequestedQuery() {
+            return requestedQuery;
```

Comment:
Does this means the query is encoded in a JSON string field AND is JSON encoded?

This seems ugly...

Maybe we can get a better result by passing the QueryDTO...